### PR TITLE
Add ARM64 Architecture Build

### PR DESCRIPTION
**Pull request by kolanos**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,14 +6,21 @@ clean:
...
-zip: dist
+dist-arm64: clean
+	env GOARCH=arm64 GOOS=linux go build -ldflags="-s -w" -o ./extensions/newrelic-lambda-extension
+	touch preview-extensions-ggqizro707
+
+zip-x86_64: dist-x86_64
+	zip -r extensions.zip preview-extensions-ggqizro707 extensions
+
+zip-arm64: dist-arm64
 	zip -r extensions.zip preview-extensions-ggqizro707 extensions
```
Comment (mrickard) [inline Makefile:21]: Did we want to give a different name to this output file to indicate the different architecture, or is that not needed?

File: Makefile [hunk 2]
```diff
@@ -6,14 +6,21 @@ clean:
...
 
-publish: zip
+publish: zip-x86_64
 	aws lambda publish-layer-version --no-cli-pager --layer-name newrelic-lambda-extension --zip-file fileb://extensions.zip
```
Comment (mrickard) [inline Makefile:24]: We'll need a different layer name for the arm64 version as well, wouldn't we?

Comment (kolanos) [reply Makefile:24]: This isn't the actual layer publishing. This was just something I believe @MattWhelan added for local testing.

File: Makefile (hunk 1 above)
Comment (kolanos) [reply Makefile:21]: It's not used currently by the CI/CD, but not a bad idea.

**Review by mrickard — APPROVED** [review]

**Pull request merged**